StdMinMaxElement.md

```diff
@@ -0,0 +1,412 @@
+auto max_element(const ExecutionSpace& exespace,                        (1)
+                 IteratorType first, IteratorType last);
+
+template <class ExecutionSpace, class IteratorType>
+auto max_element(const std::string& label,                              (2)
+                 const ExecutionSpace& exespace,
+                 IteratorType first, IteratorType last);
+
+template <class ExecutionSpace, class IteratorType, class ComparatorType>
+auto max_element(const ExecutionSpace& exespace,                        (3)
+                 IteratorType first, IteratorType last,
+                 ComparatorType comp);
+
+template <class ExecutionSpace, class IteratorType, class ComparatorType>
+auto max_element(const std::string& label,                              (4)
+                 const ExecutionSpace& exespace,
+                 IteratorType first, IteratorType last,
+                 ComparatorType comp);
+
+//
+// overload set accepting views
+//
+template <class ExecutionSpace, class DataType, class... Properties>
+auto max_element(const ExecutionSpace& exespace,                        (5)
+                 const ::Kokkos::View<DataType, Properties...>& view);
+
+template <class ExecutionSpace, class DataType, class... Properties>
+auto max_element(const std::string& label,                              (6)
+                 const ExecutionSpace& exespace,
+                 const ::Kokkos::View<DataType, Properties...>& view);
+
+template <class ExecutionSpace, class DataType, class ComparatorType, class... Properties>
+auto max_element(const ExecutionSpace& exespace,                        (7)
+                 const ::Kokkos::View<DataType, Properties...>& view,
+                 ComparatorType comp);
+
+template <class ExecutionSpace, class DataType, class ComparatorType, class... Properties>
+auto max_element(const std::string& label,                              (8)
+                 const ExecutionSpace& exespace,
+                 const ::Kokkos::View<DataType, Properties...>& view,
+                 ComparatorType comp);
+```
+
+### Description
+
+- (1,2,5,6): Finds the smallest element in the range `[first, last)` or in `view`
+  using `operator<` to compare two elements
+
+- (3,4,7,8): Finds the smallest element in the range `[first, last)` or in `view`
+  using the binary functor `comp` to compare two elements
+
+### Parameters and Requirements
+
+- `exespace`, `first`, `last`, `view`, `comp`: same as in [`min_element`](#kokkosexperimentalmin_element)
+- `label`:
+  - used to name the implementation kernels for debugging purposes
+  - for 1,3 the default string is: "Kokkos::max_element_iterator_api_default"
+  - for 5,7 the default string is: "Kokkos::max_element_view_api_default"
+
+### Return
+
+Iterator to the largest element.
+The following special cases apply:
+- if several elements are equivalent to the smallest element, returns the iterator to the *first* such element.
```

Comment:
```suggestion
- if several elements are equivalent to the largest element, returns the iterator to the *first* such element.
```

Reply:
Honestly I don't remember if it will return *first* or *last* such element.

Reply:
last one!